xbmc/epg/EpgContainer.cpp

```diff
@@ -829,13 +829,15 @@ void CEpgContainer::UpdateEpgEvents()
   if (!m_lastEpgEventPurge.IsValid() || m_lastEpgEventPurge < (now - CDateTimeSpan(1,0,0,0)))
   {
     CDateTime purgeTime = now - CDateTimeSpan(0, g_advancedSettings.m_iEpgLingerTime / 60, g_advancedSettings.m_iEpgLingerTime % 60, 0);
-    for (auto event = m_epgEvents.begin(); event != m_epgEvents.end(); ++event)
+    for (auto event = m_epgEvents.begin(); event != m_epgEvents.end();)
     {
       if (event->second->EndAsUTC() < purgeTime)
       {
         m_epgEvents.erase(event);
         ++count;
       }
+      else
+        ++event;
```

Comment:
Thanks @fritsch. You are right. I thought that <code>erase</code> would set the iterator to the first element after the erased element, mixed this up with the return value of erase. So, line 836 needs to read <code>event = m_epgEvents.erase(event);</code>, right?